packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart

```diff
@@ -54,14 +55,26 @@ class StreamVideoPushNotificationManager implements PushNotificationManager {
     this.callerCustomizationCallback,
   }) : _client = client {
     _subscriptions.add(
-      _idCallIncoming,
+      _idCallKitIncoming,
       onCallEvent.whereType<ActionCallIncoming>().listen(
         (_) {
           _subscriptions.add(
             _idCallEnded,
             client.events.on<CoordinatorCallEndedEvent>(
-              (_) {
-                endAllCalls();
+              (event) {
+                FlutterCallkitIncoming.endCall(event.callCid.id);
+              },
+            ),
+          );
+
+          _subscriptions.add(
+            _idCallAccepted,
+            client.events.on<CoordinatorCallAcceptedEvent>(
+              (event) async {
+                await FlutterCallkitIncoming.silenceEvents();
+                await FlutterCallkitIncoming.endCall(event.callCid.id);
+                await Future<void>.delayed(const Duration(milliseconds: 300));
```

Comment:
Why do we need to delay here?

Reply:
To account for asynchronous race, it happened that events were unsilenced before the endCall event was triggered on the native side.

Reply:
Fair enough.